scripts/llama_summary.py

```diff
@@ -0,0 +1,181 @@
+                },
+            ],
+            "temperature": 0.7,
+            "max_tokens": 500,
+            "model": MODEL_NAME,
+        }
+
+        # Make the API request
+        headers = {"Content-Type": "application/json", "api-key": GITHUB_TOKEN}
+
+        async with aiohttp.ClientSession() as session:
+            async with session.post(
+                GITHUB_API_ENDPOINT, json=payload, headers=headers
+            ) as response:
+                if response.status == 200:
+                    result = await response.json()
+                    if "choices" in result and len(result["choices"]) > 0:
+                        return result["choices"][0]["message"]["content"]
+                    else:
+                        print(f"Error: Unexpected response format: {result}")
+                else:
+                    error_text = await response.text()
+                    print(
+                        f"Error: API request failed with status {response.status}: {error_text}"
+                    )
+
+        # Fallback to placeholder if API call fails
+        return f"[PLACEHOLDER] This meeting ({summary.get('meeting_name')}) summary will be generated automatically."
+
+    except Exception as e:
+        print(f"Error generating summary: {e}")
+        return f"[PLACEHOLDER] Error generating summary: {str(e)}"
+
+
+async def update_summaries(summaries: List[MeetingSummary]) -> List[MeetingSummary]:
+    new_summaries = []
+
+    # Process each summary that needs updating
+    for summary in summaries:
+        print(
+            f"Generating summary for meeting: {summary.meeting.meeting} ({summary.meeting.date})"
+        )
+
+        # Generate the AI summary
+        ai_summary = await summarize_with_llama(summary)
+        # Update the summary data
+        summary.summary = ai_summary
+        summary.summarization_date = datetime.now(timezone.utc).isoformat()
+        summary.needs_summarization = False
+
+        # Update the dictionary
+        new_summaries.append(summary)
+
+    print(f"Updated {len(new_summaries)} summaries.")
+
+    # Convert back to list
+    return new_summaries
+
+
+async def save_summaries(summaries: List[MeetingSummary]) -> None:
+    """
+    Save summaries to the summaries.jsonl file.
+
+    Args:
+        summaries: List of MeetingSummary objects
+    """
+    # Save to original location
+    with SUMMARY_FILE.open("w") as f:
+        for summary in summaries:
+            # Convert the Pydantic model to a dictionary with proper serialization options
+            # By default, HttpUrl objects are not JSON serializable
+            summary_dict = summary.model_dump(mode="json")
+            f.write(json.dumps(summary_dict) + "\n")
```

Comment:
suggestion (non-blocking): this should eventually change to put these into online storage somewhere.

Reply:
Yes, I'm going to remove the jsonl creation.  I'll  await (😆) your langchain replacement